cpp/src/comms/cuML_comms_test.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cuML_comms_test.hpp"
+
+#include <common/cumlHandle.hpp>
+#include <common/cuml_comms_int.hpp>
+#include <common/device_buffer.hpp>
+#include <iostream>
+
+namespace ML {
+namespace sandbox {
+
+
+bool test_collective_allreduce(const ML::cumlHandle& h) {
```

Comment:
Since this is not part of the test infrastructure I think it would be better to call it `collective_hello_world` or something along those lines.

Reply:
This is actually part of the test infrastructure because it's being exposed to pytests through cython. This was one of the reasons I put it in a directory named `comms`. I believe this belongs in cuML and not in the comms implementation because these tests should be able to work across implementations.

Reply:
This is, perhaps, a side-effect of having split the C++ into a separate PR. 

@teju85 @jirikraus, [1] is the relevant Cython wrapper for the test and [2] contains the Pytest which uses it. I like having this in the C++ layer, rather than just using a wrapped communicator, because it allows us to test the comms API in C++. Ideally, once we have our conda MPI packaging figured, MG CI, and incorporate our MPI comm impl, these tests can be called from GTests as well.

Reply:
OK. Then I would say the sandbox namespace should be removed.